docs/en/connector-v2/source/Hive.md

```diff
@@ -0,0 +1,47 @@
+# Hive
+
+> Hive source connector
+
+## Description
+
+Read data from Hive.
+
+In order to use this connector, You must ensure your spark/flink cluster already integrated hive. The tested hive version is 2.3.9.
+
+## Key features
+
+- [x] [exactly-once](../../concept/connector-v2-features.md)
+
+By default, we use 2PC commit to ensure `exactly-once`
```

Comment:
If a source connector support `exactly-once`, it must save the split and the offset already send to down stream when snapshot.

Reply:
So file source does not support `exactly-once`, right?

Reply:
> So file source does not support `exactly-once`, right?

From the perspective of code implementation, yes.

Reply:
> So file source does not support `exactly-once`, right?

I read the code again, if we read all read of a split and send to the down stream, we can support `exactly-once`. Because the `pollNext` method and `snapshot` method use same lock `checkpointLock`.